src/CommunityPatch/Patches/Perks/Vigor/TwoHanded/EvisceratorPatch.cs

```diff
@@ -0,0 +1,117 @@
+        0xB2, 0xD3, 0x58, 0x35, 0x5B, 0x27, 0x14, 0x72
+      }
+    };
+
+    public override void Reset()
+      => _perk = PerkObject.FindFirst(x => x.Name.GetID() == "C2CwsC91");
+
+    // ReSharper disable once CompareOfFloatsByEqualityOperator
+    public override bool? IsApplicable(Game game) {
+      if (_perk == null) return false;
+      if (_perk.PrimaryBonus != 0.3f) return false;
+
+      var patchInfo = Harmony.GetPatchInfo(TargetMethodInfo);
+      if (AlreadyPatchedByOthers(patchInfo)) return false;
+
+      var hash = TargetMethodInfo.MakeCilSignatureSha256();
+      return hash.MatchesAnySha256(Hashes);
+    }
+
+    public override void Apply(Game game) {
+      var textObjStrings = TextObject.ConvertToStringList(
+        new List<TextObject> {
+          _perk.Name,
+          _perk.Description
+        }
+      );
+      // most of the properties of skills have private setters, yet Initialize is public
+      _perk.Initialize(
+        textObjStrings[0],
+        textObjStrings[1],
+        _perk.Skill,
+        (int) _perk.RequiredSkillValue,
+        _perk.AlternativePerk,
+        _perk.PrimaryRole, -30f,
+        _perk.SecondaryRole, _perk.SecondaryBonus,
+        SkillEffect.EffectIncrementType.AddFactor
+      );
+      if (Applied) return;
+
+      CommunityPatchSubModule.Harmony.Patch(TargetMethodInfo, new HarmonyMethod(PatchMethodInfo));
+      Applied = true;
+    }
+
+    // ReSharper disable once InconsistentNaming
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static bool Prefix(AgentMoraleInteractionLogic __instance, Agent affectedAgent, Agent affectorAgent, AgentState agentState) {
+      var affectorCharacter = (CharacterObject) affectorAgent?.Character;
+      
+      if (affectorCharacter == null) return false;
+      if (affectedAgent.Character == null) return false;
+      if (agentState != AgentState.Killed && agentState != AgentState.Unconscious) return false;
+      if (affectedAgent.Team == null) return false;
+      
+      var moralChangesTuple = MissionGameModels.Current.BattleMoraleModel.CalculateMoraleChangeAfterAgentKilled(affectedAgent);
+      var moralChangeFriend = moralChangesTuple.Item1;
+      var moralChangeEnemy = moralChangesTuple.Item2;
+      
+      if (!AnyMoralChanges(moralChangeEnemy, moralChangeFriend)) return false;
+
+      if (affectorCharacter.GetPerkValue(ActivePatch._perk)) 
+        moralChangeFriend += moralChangeFriend * ActivePatch._perk.PrimaryBonus / -100;
+      
+      __instance.ApplyAoeMoraleEffect(
+        affectedAgent.GetWorldPosition(), 
+        affectorAgent.GetWorldPosition(), 
+        affectedAgent.Team, 
+        moralChangeFriend, 
+        moralChangeEnemy, 
+        10f, 
+        a => a.IsActive() && a.Formation != null && a.Formation == affectedAgent.Formation, 
+        a => a.IsActive() && a.Formation != null && a.Formation == affectorAgent.Formation);
+
+      return false;
+    }
```

Comment:
But there doesn't seem to be any good alternative here.

Reply:
Well, you could register a new `public class EvisceratorAgentMoraleInteractionLogic : MissionLogic`, have it override `OnAgentRemoved` and register it to some behavior injection somewhere... You would do everything the same in the method except just work with the fraction of morale instead of the total. It's probably more effort than it's worth.

Let's stick with the Prefix replacing the method for now.